framework/aster-frame/src/task/processor.rs

```diff
@@ -117,9 +119,39 @@ fn switch_to_task(next_task: Arc<Task>) {
 
     let next_task_cx_ptr = &next_task.inner_ctx() as *const TaskContext;
 
+    // If the current and next tasks meet the following conditions,
+    // then we can safely skip switching page table for improved performance.
+    //
+    // Current task         | Next task                 | Should switch page table?
+    // ----------------------------------------------------------------------------
+    // any                  | w/o a user space          | No
+    // w/ a user space.     | w/ the same user space    | No
```

Comment:
I think the current approach is acceptable. However, I would like to ask, why not do this optimization at `VmSpace::activate`? We can load the register pointing to the page table root (`cr3` on x86), check if the page table root is the same as the `VmSpace` we're trying to activate, and only trigger the page table reload if the answer is negative.

My feeling is that the overhead of doing the check in `switch_to_task()` or doing the check in `VmSpace::activate` is comparable. However, the latter approach can additionally avoid page table reloads in the case where the context switches look like `user task A -> kernel task B -> user task A`.

Changing the semantics of `VmSpace::activate` to "activate if not already activated" makes a lot of sense to me, and if we really want to force-activate a `VmSpace` (to flush TLB?) we can always have another method.

Reply:
Your suggestion makes sense. And someone can do it after #713 is merged. Or perhaps I can immediately do it now.

Reply:
Agree. Putting the two optimizations in two places makes sense.

Reply:
Agree

Reply:
Done